src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,292 @@
+package com.cloudant.client.api;
+
+import static com.cloudant.client.org.lightcouch.internal.CouchDbUtil.assertNotEmpty;
+
+import com.cloudant.client.org.lightcouch.CouchDbException;
+import com.cloudant.client.org.lightcouch.CouchDbProperties;
+import com.cloudant.client.org.lightcouch.internal.URIBuilder;
+import com.cloudant.http.CookieInterceptor;
+import com.cloudant.http.HttpConnectionInterceptor;
+import com.cloudant.http.HttpConnectionRequestInterceptor;
+import com.cloudant.http.HttpConnectionResponseInterceptor;
+import com.cloudant.http.interceptors.ProxyAuthInterceptor;
+import com.cloudant.http.interceptors.SSLCustomizerInterceptor;
+import com.cloudant.http.interceptors.TimeoutCustomizationInterceptor;
+import com.google.gson.GsonBuilder;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.SSLSocketFactory;
+
+public class ClientBuilder {
+
+     /** Default of 5 minutes for connection timeout **/
+    public static final long CONNECTION_TIMEOUT = 5l;
+    /** Default of 5 minutes for read timeout **/
+    public static final long READ_TIMEOUT = 5l;
+
+    /** Default max of 6 connections **/
+    public static final int MAX_CONNECTIONS = 6;
+    /** Default timeout for read and connect timeout **/
+    private static final TimeoutOption DEFAULT_CONNECTION_TIMEOUT
+            = TimeoutOption.minutes(CONNECTION_TIMEOUT);
+    private static final TimeoutOption DEFAULT_READ_TIMEOUT
+            = TimeoutOption.minutes(READ_TIMEOUT);
+
+    private List<HttpConnectionRequestInterceptor> requestInterceptors = new ArrayList
+            <HttpConnectionRequestInterceptor>();
+    private List<HttpConnectionResponseInterceptor> responseInterceptors = new ArrayList
+            <HttpConnectionResponseInterceptor>();
+    private String password;
+    private String username;
+    private URL url;
+    private GsonBuilder gsonBuilder;
+    /** Defaults to {@link #MAX_CONNECTIONS} **/
+    private int maxConnections = MAX_CONNECTIONS;
+    private URL proxyURL;
+    private String proxyUser;
+    private String proxyPassword;
+    private boolean isSSLAuthenticationDisabled;
+    private SSLSocketFactory authenticatedModeSSLSocketFactory;
+    /** Defaults to {@link #CONNECTION_TIMEOUT} **/
+    private TimeoutOption connectionTimeout = DEFAULT_CONNECTION_TIMEOUT;
+    /** Defaults to {@link #READ_TIMEOUT} **/
+    private TimeoutOption readTimeout = DEFAULT_READ_TIMEOUT;
+
+
+    public static ClientBuilder account (String account) {
+        return new ClientBuilder(account);
+    }
+
+    public static ClientBuilder url(URL url) {
+        return new ClientBuilder(url);
+    }
+
+    public ClientBuilder(String account) {
+        //parse account into a url
+        assertNotEmpty(account, "accountName");
+        try {
+            this.url = URIBuilder.buildUri()
```

Comment:
Should we make this URL then call `return url(url)`? It doesn't offer huge advantages, but means we can have the exception handling in just one place, which might be easier if we change it later.
I guess we also wouldn't need to hardcode the port because we would get the default for `https` too.